### ISSUE TITLE
[Bug] Not all parsing errors are reported

### ISSUE THREAD

**Issue by bhirsz** [MarketSquare/robotframework-robocop#461]
Some parsing errors are not reporting - investigate why. 

From #460

**Comment by bhirsz** [comment]
Changing `visit_Error` to `visit_Statement` fix it. I've got 80 instead of 40 parsing errors when processing our tests directory. I see that some of the issues reported as parsing-error are already catched by other rules. Should be filter our parsing-error (and introduce new rules based on that)? For example we should catch this:
```
0401 Robot Framework syntax error: Invalid variable name '... a'. (parsing-error)
```
and report it as not enough whitespace after continuation marker.

**Comment by bhirsz** [comment]
And some of the issues that are already catched by other rule and then also by parsing-error:
```
0401 Robot Framework syntax error: Invalid variable name ' ${var4}'. (parsing-error)
1014 Variable in variable section should be left aligned. (variable-should-left-aligned)
```

**Comment by mnojek** [comment]
Hmm... I think it would be better to catch these errors and parse it to Robocop style instead of blindly reporting RF errors. Especially when we already have rules that catch similar or the same errors inside Robocop.

**Comment by pekkaklarck** [comment]
The root cause is that not all errors are reported as error tokens or nodes. For example, FOR loop without an end will be a FOR loop in AST but the node has `errors` attribute that contains the error (or errors).

**Comment by bhirsz** [comment]
> The root cause is that not all errors are reported as error tokens or nodes. For example, FOR loop without an END will be a FOR loop in AST but the node has `errors` attribute that contains the error (or errors).

Yeah - noticed that and we will just go through all nodes now and check errors attr instead of relying only on error tokens.
